packages/expo-file-system/android/src/main/java/expo/modules/filesystem/next/FileSystemDirectory.kt

```diff
@@ -29,6 +30,33 @@ class FileSystemDirectory(file: File) : FileSystemPath(file) {
     return file.walkTopDown().filter { it.isFile }.map { it.length() }.sum()
   }
 
+  fun info(): DirectoryInfo {
+    validateType()
+    validatePermission(Permission.READ)
+    if (!file.exists()) {
+      val directoryInfo = DirectoryInfo(
+        exists = false,
+        uri = slashifyFilePath(file.toURI().toString())
+      )
+      return directoryInfo
+    }
+
+    when {
+      file.toURI().scheme == "file" -> {
+        val directoryInfo = DirectoryInfo(
+          exists = true,
+          uri = slashifyFilePath(file.toURI().toString()),
+          files = file.listFiles()?.map { i -> i.name },
+          modificationTime = modificationTime,
+          creationTime = creationTime,
+          size = size
+        )
+        return directoryInfo
+      }
+      else -> throw UnableToGetInfoException("file schema ${file.toURI().scheme} is not supported")
```

Comment:
the error message could say that only the `file` scheme is supported.

This gets me wondering: what schemes, other than `file`, would pass the `file.exists()` test?